aks-node-controller/app.go

```diff
@@ -152,7 +171,7 @@ func (a *App) ProvisionWait(ctx context.Context, filepaths ProvisionStatusFiles)
 			if event.Op&fsnotify.Create == fsnotify.Create && event.Name == filepaths.ProvisionCompleteFile {
 				data, err := os.ReadFile(filepaths.ProvisionJSONFile)
 				if err != nil {
-					return "", fmt.Errorf("failed to read provision.json: %w", err)
+					return "", fmt.Errorf("failed to read provision.json: %w. One reason could be that AKSNodeConfig is not properly set", err)
```

Comment:
I don't think this extra message is helpful.

`fmt.Errorf("failed to read %s: %w", filepaths.ProvisionJSONFile, err)` should be good enough?

Reply:
btw, provision.json is provision scripts output, not config.json (which is the aksnodeconfig)

Reply:
Later I will write a wiki to help debugging